test/test_katello/test_deb_tracer.py

```diff
@@ -0,0 +1,73 @@
+import unittest2 as unittest
+
+# sys.path.append(os.path.join(os.path.dirname(__file__), '../../src/'))
+# from katello.tracer import upload_tracer_profile, query_affected_apps
+# from katello.constants import YUM, ZYPPER
+from katello.deb_tracer import needrestart, has_needrestart
+import subprocess
+
+from mock import patch, Mock
+
+import pytest
+
+class TestNeedrestart(unittest.TestCase):
+    tracer_output = """ perl: warning: Setting locale failed.
+perl: warning: Please check that your locale settings:
+	LANGUAGE = (unset),
+	LC_ALL = (unset),
+	LC_ADDRESS = "fr_FR.UTF-8",
+	LC_NAME = "fr_FR.UTF-8",
+	LC_MONETARY = "fr_FR.UTF-8",
+	LC_PAPER = "fr_FR.UTF-8",
+	LC_IDENTIFICATION = "fr_FR.UTF-8",
+	LC_TELEPHONE = "fr_FR.UTF-8",
+	LC_MEASUREMENT = "fr_FR.UTF-8",
+	LC_TIME = "fr_FR.UTF-8",
+	LC_NUMERIC = "fr_FR.UTF-8",
+	LANG = "en_US.UTF-8"
+    are supported and installed on your system.
+perl: warning: Falling back to a fallback locale ("en_US.UTF-8").
```

Comment:
The mock data should only contain the real output - not the locale warnings.

Reply:
I also think that goes to stderr, which is a good point to think about. https://docs.python.org/3/library/subprocess.html#subprocess.check_output says `stderr` is discarded by default and can be captured optionally. So to capture it manually, I'd run `needrestart -b 2> /dev/null`.